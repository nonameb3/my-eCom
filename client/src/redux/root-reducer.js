import { combineReducers } from "redux";
import { persistReducer } from "redux-persist";
import storage from "redux-persist/lib/storage";

import userReducer from "./user/user.reducer";
import cartReducer from "./cart/cart.reducer";
import directoryReducer from "./directory/directory.reducer";
import shopReducer from "./shop/shop.reducer";

// session store config
const persistConfig = {
  key: "root",
  storage: storage,
  whitelist: ["cart"]
};

// reducer
const rootReducer = combineReducers({
  user: userReducer,
  cart: cartReducer,
  directory: directoryReducer,
  shop: shopReducer
});

// reducer with session store
export default persistReducer(persistConfig, rootReducer);
