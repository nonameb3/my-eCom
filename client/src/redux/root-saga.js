import { all, call } from "redux-saga/effects";

import { shopSaga } from "./shop/shop.sagas";
import { userSagas } from "./user/user.saga";
import { cartSaga } from "./cart/cart.saga";

export default function* rootSaga() {
  yield all([call(shopSaga), call(userSagas), call(cartSaga)]);
}
