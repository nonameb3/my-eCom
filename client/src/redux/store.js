import { createStore, applyMiddleware } from "redux";
import { persistStore } from "redux-persist";
import logger from "redux-logger";
import createReduxSagamiddelware from "redux-saga";

import rootSaga from "./root-saga";
import rootReducer from "./root-reducer";

const sagaMiddelware = createReduxSagamiddelware();
let middlewares = [sagaMiddelware];
if (process.env.NODE_ENV === "development") {
  middlewares = [...middlewares, logger];
}

export const store = createStore(rootReducer, applyMiddleware(...middlewares));
sagaMiddelware.run(rootSaga);

export const persistor = persistStore(store);
