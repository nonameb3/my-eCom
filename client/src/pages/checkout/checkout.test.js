import React from "react";
import { shallow } from "enzyme";
import { Provider } from "react-redux";

import Checkout from "./checkout.component";
import { store } from "../../redux/store";

let wrapper;
beforeEach(() => {
  const mockData = {
    cartItems: [],
    total: 99
  };

  wrapper = shallow(
    <Provider store={store}>
      <Checkout {...mockData} />
    </Provider>
  );
});

it("should render checkout component", () => {
  expect(wrapper).toMatchSnapshot();
});
