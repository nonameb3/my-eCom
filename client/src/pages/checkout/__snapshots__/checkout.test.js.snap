// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render checkout component 1`] = `ShallowWrapper {}`;
